File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minprice = Integer.MAX_VALUE; // Initialize minprice to the highest possible value
        int maxprofit = 0;                // Initialize maxprofit to 0 (no profit)

        for (int i = 0; i < prices.length; i++) {
            // Update minprice if the current price is less than minprice
            if (prices[i] < minprice) {
                minprice = prices[i];
            } 
            
            // Calculate the profit by selling on the current day and compare with maxprofit
            int profit = prices[i] - minprice;
            if (profit > maxprofit) {
                maxprofit = profit;
            }
        }
        return maxprofit; // Return the maximum profit found
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprices=prices[0]
        minp=0
        for i in range(len(prices)):
            if prices[i]<minprices:
                minprices=prices[i]
            if prices[i]-minprices>minp:
                minp=prices[i]-minprices
        return minp

       
        